src/Sylius/Component/Core/Taxation/Processor/OrderTaxesProcessor.php

```diff
@@ -86,18 +86,20 @@ public function process(OrderInterface $order)
     }
 
     /**
-     * @param AddressInterface|null $shippingAddress
+     * @param OrderInterface $order
      *
      * @return ZoneInterface|null
      */
-    private function getTaxZone(AddressInterface $shippingAddress = null)
+    private function getTaxZone(OrderInterface $order)
     {
+        $shippingAddress = $order->getShippingAddress();
         $zone = null;
+
         if (null !== $shippingAddress) {
             $zone = $this->zoneMatcher->match($shippingAddress);
```

Comment:
Why don't we wrap this zone matcher with `TaxZoneProviderInterface`? Then we can setup a composite one, that checks this one first and fallback to channel based tax zone provider, so this method would not be needed (we'll just use `$taxZoneProvider->getZone($order)` to get tax zone for given order).